### PR TITLE
Freeze requirements with pip-tools

### PR DESCRIPTION
**Pull request by lfdebrux**
I've been having issues with dmrunner and the latest version of digitalmarketplace-developer-tools, I think this is required to get dmrunner properly working with it.

Also it makes this repo more similar to our other repos, which is good.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -21,7 +21,10 @@ jobs:
         key: venv-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/requirements*.txt') }}
 
     - name: Install python dependencies
-      run: make install
+      run: |
+        python -m venv venv
```
Comment (gidsg) [inline .github/workflows/test.yml:25]: Is there a reason not to use `make virtualenv` or to make this change in the `Makefile`?

Comment (lfdebrux) [reply .github/workflows/test.yml:25]: Oh yeah, I was just being lazy because I'm going to replace the Makefile in a following PR 😅

Comment (lfdebrux) [reply .github/workflows/test.yml]: Changed the Makefile instead now

**Review by gidsg — APPROVED** [review]
Works for me locally and clearly CI is happy 😄 👍

**Pull request merged**